pkg/arvo/app/chat.hoon

```diff
@@ -341,195 +253,188 @@
-    =/  peers/(set @p)
-      ?:  =(%remove -.dif.sto)
-        (~(del in u.upeers) who.sto)
-      (~(put in u.upeers) who.sto)
-    =/  str
-      %=  str.sta
-        peers  (~(put by peers.str.sta) cir.sto peers)
+    =/  sta
+      %=  str
+        configs  (~(put by configs.str) circ `conf)
       ==
-    :-  (send-chat-update [[%peers cir.sto peers] str])
-    this(str.sta str)
-    ::
-    ::  %config: config has changed
-    ::
-        %config
-    =*  circ  cir.sto
-    ::
-    ?+  -.dif.sto
+    :_  this(str sta)
+    (send-chat-update [[%config circ conf] sta])
+  ::
+  ::  %source: the sources of our inbox have changed
+  ::
+      %source
+    ?.  =(circ [our.bol %inbox])
+      :: ignore when circ is not inbox
       [~ this]
-      ::
-      ::  %full: set all of config without side effects
-      ::
-          %full
-        =*  conf  cof.dif.sto
-        =/  str
-          %=  str.sta
-            configs  (~(put by configs.str.sta) circ `conf)
-          ==
-        :-  (send-chat-update [[%config circ conf] str])
-        this(str.sta str)
-      ::
-      ::  %read: the read count of one of our configs has changed
-      ::
-          %read
-        ?:  =(circ [our.bol %inbox])
-          ::  ignore when circ is inbox
-          [~ this]
-        =/  uconf/(unit config:hall)  (~(got by configs.str.sta) circ)
-        ?~  uconf
-          ::  should we crash?
-          [~ this]
-        =/  conf/config:hall
-          %=  u.uconf
-            red  red.dif.sto
-          ==
-        =/  str
-          %=  str.sta
-            configs  (~(put by configs.str.sta) circ `conf)
-          ==
-        :-  (send-chat-update [[%config circ conf] str])
-        this(str.sta str)
-      ::
-      ::  %source: the sources of our inbox have changed
-      ::
-          %source
-        ?.  =(circ [our.bol %inbox])
-          :: ignore when circ is not inbox
-          [~ this]
-        =*  affectedcir  cir.src.dif.sto
-        =/  newwir/wire
-          /circle/[(scot %p hos.affectedcir)]/[nom.affectedcir]/grams/0/config/group
-        =/  pat/path  /circle/[nom.affectedcir]/grams/0/config/group
-        ::  we've added a source to our inbox
-        ::
-        ?:  add.dif.sto
-          =/  newinbox  %=  inbox.str.sta
-            src  (~(put in src.inbox.str.sta) src.dif.sto)
-          ==
-          =/  str
-            %=  str.sta
-              inbox  newinbox
-              ::
-              configs
-                ?:  (~(has by configs.str.sta) affectedcir)
-                  configs.str.sta
-                (~(put by configs.str.sta) affectedcir ~)
-            ==
+    =*  affectedcir  cir.src.dif.sto
+    =/  newwir/wire
+      /circle/(scot %p hos.affectedcir)/[nom.affectedcir]/grams/0/config/group
+    =/  pat/path  /circle/[nom.affectedcir]/grams/0/config/group
+    ::  we've added a source to our inbox
+    ::
+    ?:  add.dif.sto
+      =/  newinbox  %=  inbox.str
+        src  (~(put in src.inbox.str) src.dif.sto)
```

Comment:
Either one-line it entirely, or split it up properly. The `%=` can probably be a one-liner regardless though.